test/spec/todo/todo.spec.ts

```diff
@@ -0,0 +1,197 @@
+require ('module-alias/register');
+import chai from 'chai';
+import chaiHttp from 'chai-http';
+import { Application } from 'express';
+import { TodoItem } from '../../../src/models';
+import { App } from '../../../src/server';
+import { respositoryContext, testAppContext } from '../../mocks/app-context';
+
+
+chai.use(chaiHttp);
+const expect = chai.expect;
+let expressApp: Application;
+
+before(async () => {
+  //await respositoryContext.store.connect();
+  const app = new App(testAppContext);
+  
+
+  app.initializeMiddlewares();
+  app.initializeControllers();
+  app.initializeErrorHandling();
+
+  expressApp = app.expressApp;
+});
+
+describe('POST /todo', () => {
+  it('should create a new todo item', async () => {
+    const res = await chai.request(expressApp).post('/todos').send({
+      title: 'Todo Item',
+    });
+
+    expect(res).to.have.status(201);
+    expect(res.body).to.have.property('id');
+    expect(res.body).to.have.property('title');
+  });
+
+  it('should not allow blank title', async () => {
+    const res = await chai.request(expressApp).post('/todos').send({
+      title: '',
+    });
+
+    expect(res).to.have.status(400);
+    expect(res.body).to.have.property('message');
+    expect(res.body)
+      .to.have.nested.property('failures[0].message')
+      .to.equal('VALIDATION_ERRORS.INVALID_TITLE');
+  });
+
+  it('should not allow duplicate title', async () => {
+    await testAppContext.todoRepository.save(
+      new TodoItem({
+        title: 'Duplicate Check Todo Item',
+      })
+    );
+
+    const res = await chai.request(expressApp).post('/todos').send({
+      title: 'Duplicate Check Todo Item',
+    });
+
+    expect(res).to.have.status(400);
+    expect(res.body).to.have.property('message');
+    expect(res.body)
+      .to.have.nested.property('failures[0].message')
+      .to.equal('VALIDATION_ERRORS.DUPLICATE_ENTRY');
+  });
+});
+
+describe('DELETE /todos/:id', () => {
+  it('should delete a todo item if it exists and if ID it is valid.', async () => {
+    const todoItem = await testAppContext.todoRepository.save(
+      new TodoItem({ title: 'Title to be deleted' })
+    );
+    const res = await chai.request(expressApp).delete(`/todos/${todoItem._id}`);
+
+    expect(res).to.have.status(204);
```

Comment:
We should also try to fetch TODO from database for the same TODO id and make sure it does not exists anymore. There can be cases sometimes when deleting an entity via API call returns success but the entity still exists in the database.